content/blog/mcp-with-wasmcp.md

```diff
@@ -0,0 +1,387 @@
+                        },
+                        "required": ["a", "b"]
+                    }"#
+                    .to_string(),
+                    options: None,
+                },
+                Tool {
+                    name: "subtract".to_string(),
+                    input_schema: r#"{
+                        "type": "object",
+                        "properties": {
+                            "a": {"type": "number", "description": "Number to subtract from"},
+                            "b": {"type": "number", "description": "Number to subtract"}
+                        },
+                        "required": ["a", "b"]
+                    }"#
+                    .to_string(),
+                    options: None,
+                },
+            ],
+            next_cursor: None,
+            meta: None,
+        })
+    }
+
+    fn call_tool(
+        _ctx: RequestCtx,
+        request: CallToolRequest,
+    ) -> Result<Option<CallToolResult>, ErrorCode> {
+        match request.name.as_str() {
+            "add" => Ok(Some(execute_operation(&request.arguments, |a, b| a + b))),
+            "subtract" => Ok(Some(execute_operation(&request.arguments, |a, b| a - b))),
+            _ => Ok(None), // We don't handle this tool
+        }
+    }
+}
+```
+
+Now let’s build this component and compose it into a full MCP server. The `wasmcp compose server` command
+1. Pulls the default [wasmcp framework components](https://github.com/wasmcp/wasmcp/tree/main/crates), like the MCP transport and related plumbing, from [GitHub Container Registry](https://github.com/orgs/wasmcp/packages?repo_name=wasmcp).
+2. Plugs your component binary into the wasmcp framework components, producing a complete `server.wasm` component.
+
+This is accomplished with Bytecode Alliance’s [wac](https://github.com/bytecodealliance/wac) tooling, which you can also use directly for composition.
+
+Note that any of the framework-level components can also be interchanged with your own custom implementations, like a custom transport component. See `wasmcp compose server –help` for details.
+
+```
+$ make
+$ wasmcp compose server target/wasm32-wasip2/release/my-first-tools.wasm -o server.wasm
+```
+
+Now that we have a complete `server.wasm` component, we can run it directly with `spin up`.
+
+```
+$ spin up -f server.wasm
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  http-trigger1-component: http://127.0.0.1:3000 (wildcard)
+```
+
+Note that runtime configuration can be managed with a [spin.toml](https://spinframework.dev/v3/writing-apps) file.
+
+And _just like that_, we have a functional MCP server over the Streamable HTTP transport! Authorization for providers that implement Dynamic Client Registration is configurable via [environment variables](https://github.com/wasmcp/wasmcp/tree/main/docs). The `stdio` transport can also be used via [Wasmtime](https://github.com/bytecodealliance/wasmtime) directly.
```

Comment:
I think this is a great opportunity to talk about the cross platform nature of components. Maybe mention way we can also use wasmtime serve here -- because the component exposes `wasi:http/incoming-handler` which both runtimes can serve